arch/risc-v/src/common/riscv_idle.c

```diff
@@ -74,15 +74,6 @@ void up_idle(void)
    * sleep in a reduced power mode until an interrupt occurs to save power
    */
 
-  /* This is a kludge that I still don't understand.  The call to
```

Comment:
@xiaoxiang781216 did you try to remove the "kludge"?
It's this part:
```
#ifdef CONFIG_SCHED_WORKQUEUE
  irqstate_t flags = enter_critical_section();
  leave_critical_section(flags);
#endif
```
This also could be another thing that came from the MIPS port.  We had this bug in MIPS where an interrupt mask wasn't set correctly and disabling/enabling interrupts was just clearing that mask, now it's fixed.

RISC-V chips are using a custom idle loop that doesn't include this kludge.

Reply:
Done, and remove other arch too, because I don't believe the empty critical section can have the real help.